src/cargo/ops/resolve.rs

```diff
@@ -116,7 +116,8 @@ version. This may also occur with an optional dependency that is not enabled.";
 /// This is a simple interface used by commands like `clean`, `fetch`, and
 /// `package`, which don't specify any options or features.
 pub fn resolve_ws<'a>(ws: &Workspace<'a>, dry_run: bool) -> CargoResult<(PackageSet<'a>, Resolve)> {
-    let mut registry = PackageRegistry::new(ws.gctx())?;
+    let source_config = SourceConfigMap::new_with_overlays(ws.gctx(), ws.local_overlays()?)?;
+    let mut registry = PackageRegistry::new_with_source_config(ws.gctx(), source_config)?;
```

Comment:
How do we decide when to replace `PackageRegistry::new` with this?

We only replaced two uses of it:
```
src/cargo/ops/resolve.rs
119:    let mut registry = PackageRegistry::new(ws.gctx())?;
145:    let mut registry = PackageRegistry::new(ws.gctx())?;

src/cargo/ops/cargo_package.rs
475:    let mut tmp_reg = PackageRegistry::new(ws.gctx())?;

src/cargo/ops/cargo_update.rs
35:    let mut registry = PackageRegistry::new(ws.gctx())?;
76:                    let mut registry = PackageRegistry::new(opts.gctx)?;
91:    let mut registry = PackageRegistry::new(opts.gctx)?;
229:    let mut registry = PackageRegistry::new(gctx)?;

src/cargo/ops/cargo_add/mod.rs
81:    let mut registry = PackageRegistry::new(options.gctx)?;

crates/xtask-bump-check/src/xtask.rs
393:    let mut registry = PackageRegistry::new(gctx)?;
```